src/common/Requirements.ts

```diff
@@ -103,73 +110,63 @@ export function WrappedPromise(
         });
 }
 
-export abstract class BaseSetup implements RequirementList {
-    public skipBaseRequirements: boolean = false;
-    public skipAdditionalRequirements: boolean = false;
-    public requirements: Requirement[];
-    public additionalRequirements: Requirement[];
-    public logger: Logger;
-    public setupMessages = Messages.loadMessages(
-        '@salesforce/lwc-dev-mobile-core',
-        'setup'
-    );
-
-    constructor(logger: Logger) {
-        this.logger = logger;
-        this.requirements = [];
-        this.additionalRequirements = [];
-    }
+const messages = Messages.loadMessages(
+    '@salesforce/lwc-dev-mobile-core',
+    'requirement'
+);
 
+export class RequirementProcessor {
     /**
-     * Executes all of the base and additional requirement steps.
-     * @returns A SetupTestResult containing all of the results of the requirement tests
+     * Executes all of the base and command requirement checks.
      */
-    public async executeSetup(): Promise<SetupTestResult> {
-        const testResult: SetupTestResult = {
+    public static async execute(requirements: {
+        [key: string]: RequirementList;
+    }): Promise<void> {
+        const testResult: RequirementCheckResult = {
             hasMetAllRequirements: true,
             tests: []
         };
 
         let totalDuration = 0;
         let allRequirements: Requirement[] = [];
```

Comment:
I'd rename this to `enabledRequirements`. It's actually misleading now to be called `allRequirements`.